modules/Kconfig.tfm

```diff
@@ -5,7 +5,13 @@
 
 menuconfig BUILD_WITH_TFM
 	bool "Build with TF-M as the Secure Execution Environment"
-	depends on TRUSTED_EXECUTION_NONSECURE
+	select CMSIS_RTOS_V2
+	select POLL
+	select THREAD_NAME
+	select THREAD_STACK_INFO
+	select INIT_STACKS
+	select THREAD_MONITOR
+	select TRUSTED_EXECUTION_NONSECURE
```

Comment:
I am not sure this is correct (need to think, at least)
I think the opposite: BUILD_WITH_TFM shall depend on TRUSTED_EXECUTION_NONSECURE.

TRUSTED_EXECUTION_NONSECURE comes directly from the board -ns- definition, and shall switch in all non-secure stuff (including the ability to build with TFM)

Reply:
The user selects to build a Non-Secure Zephyr application by defining an "ns" version of a build target. This currently sets CONFIG_TRUSTED_EXECUTION_NONSECURE=y.  You can build a non-secure Zephyr with or without TFM support. That's why I thin the reverse is more natural dependency order.

Reply:
I agree, will fix

Reply:
Done